### PR TITLE
New hub onboarding/setup flow

### PR DESCRIPTION
**Pull request by jits**
**TODO:**

- [X] Redirect users to this new page when they log in, if they haven't already completed it
- ~~Allow platform theme resources to be scoped based on the user's new scope~~ _will do a separate PR_
- ~~Allow support request templates to be scoped based on the user's new scope~~ _will do a separate PR_

### CODE REVIEW

File: platform-hub-api/app/models/user.rb [hunk 1]
```diff
@@ -6,6 +6,8 @@ class User < ApplicationRecord
   audited descriptor_field: :email
   has_associated_audits
 
+  before_create :build_flags
```
Comment (marcinc) [inline platform-hub-api/app/models/user.rb]: I might be missing something but I don't think this callback is defined?

File: platform-hub-api/app/models/user.rb [hunk 2]
```diff
@@ -21,6 +23,12 @@ class User < ApplicationRecord
 
   has_many :projects, through: :memberships
 
+  has_one :flags,
```
Comment (marcinc) [inline platform-hub-api/app/models/user.rb]: `has_many`?

File: platform-hub-api/app/serializers/user_serializer.rb [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
-
-  attributes :enabled_identities
+  has_many :identities, if: :is_admin_or_own?
 
-  def enabled_identities
+  attribute :enabled_identities do
     object.identities.pluck(:provider)
   end
+
+  has_one :flags, if: :is_admin_or_own?, serializer: UserFlagsSerializer do
```
Comment (marcinc) [inline platform-hub-api/app/serializers/user_serializer.rb]: `has_many`?

File: platform-hub-api/spec/factories/user_flags.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :user_flags do
+
```
Comment (marcinc) [inline platform-hub-api/spec/factories/user_flags.rb]: Body missing.

File: platform-hub-api/app/models/user.rb (hunk 1 above)
Comment (jits) [inline platform-hub-api/app/models/user.rb]: This uses a method provided by the `has_one` association (figured it's easier to reuse that directly than wrap it in another method).

File: platform-hub-api/app/models/user.rb (hunk 2 above)
Comment (jits) [inline platform-hub-api/app/models/user.rb]: A user has one flags object (i.e one record in the `UserFlags` table), as discussed a couple of days ago :)

File: platform-hub-api/app/serializers/user_serializer.rb (hunk 3 above)
Comment (jits) [inline platform-hub-api/app/serializers/user_serializer.rb]: (see reply on previous comment)

File: platform-hub-api/spec/factories/user_flags.rb (hunk 4 above)
Comment (jits) [inline platform-hub-api/spec/factories/user_flags.rb]: Right now this factory doesn't need anything initialised; will delete it.

File: platform-hub-api/app/models/user.rb (hunk 2 above)
Comment (marcinc) [inline platform-hub-api/app/models/user.rb]: Ah, true :)

File: platform-hub-api/app/models/user.rb (hunk 1 above)
Comment (marcinc) [inline platform-hub-api/app/models/user.rb]: OK

File: platform-hub-api/spec/factories/user_flags.rb (hunk 4 above)
Comment (marcinc) [inline platform-hub-api/spec/factories/user_flags.rb]: 👍

**Pull request merged**